test/integration/_bridging-non-rebasable.ts

```diff
@@ -0,0 +1,714 @@
+        l2TotalSupplyBefore.sub(withdrawalAmount)
+      );
+    })
+
+    .step("Finalize withdrawal on L1", async (ctx) => {
+      const {
+        l1Token,
+        l1CrossDomainMessenger,
+        l1LidoTokensBridge,
+        l2CrossDomainMessenger,
+        l2Token,
+        l2ERC20ExtendedTokensBridge,
+      } = ctx;
+      const {
+        accountA: tokenHolderA,
+        accountB: tokenHolderB,
+        l1Stranger,
+      } = ctx.accounts;
+      const { depositAmount, withdrawalAmount } = ctx.constants;
+
+      const tokenHolderABalanceBefore = await l1Token.balanceOf(
+        tokenHolderA.address
+      );
+      const l1ERC20ExtendedTokensBridgeBalanceBefore = await l1Token.balanceOf(
+        l1LidoTokensBridge.address
+      );
+
+      await l1CrossDomainMessenger
+        .connect(l1Stranger)
+        .setXDomainMessageSender(l2ERC20ExtendedTokensBridge.address);
+
+      const tx = await l1CrossDomainMessenger
+        .connect(l1Stranger)
+        .relayMessage(
+          l1LidoTokensBridge.address,
+          l2CrossDomainMessenger.address,
+          l1LidoTokensBridge.interface.encodeFunctionData(
+            "finalizeERC20Withdrawal",
+            [
+              l1Token.address,
+              l2Token.address,
+              tokenHolderB.address,
+              tokenHolderA.address,
+              withdrawalAmount,
+              "0x",
+            ]
+          ),
+          0
+        );
+
+      await assert.emits(l1LidoTokensBridge, tx, "ERC20WithdrawalFinalized", [
+        l1Token.address,
+        l2Token.address,
+        tokenHolderB.address,
+        tokenHolderA.address,
+        withdrawalAmount,
+        "0x",
+      ]);
+
+      const l1LidoTokensBridgeBalanceAfter = await l1Token.balanceOf(l1LidoTokensBridge.address);
+      const tokenHolderABalanceAfter = await l1Token.balanceOf(tokenHolderA.address);
+      const tokenHolderBBalanceAfter = await l2Token.balanceOf(tokenHolderB.address);
+
+      assert.equalBN(
+        l1LidoTokensBridgeBalanceAfter,
+        l1ERC20ExtendedTokensBridgeBalanceBefore.sub(withdrawalAmount)
+      );
+
+      assert.equalBN(
+        tokenHolderABalanceAfter,
+        tokenHolderABalanceBefore.add(withdrawalAmount)
+      );
+
+      /// check that user balance is correct after depositing and withdrawal.
+      const deltaDepositWithdrawal = depositAmount.sub(withdrawalAmount);
+      assert.equalBN(
+        ctx.balances.accountABalanceBeforeDeposit,
+        tokenHolderABalanceAfter.add(deltaDepositWithdrawal)
+      );
+      assert.equalBN(
+        ctx.balances.accountBBalanceBeforeDeposit,
+        tokenHolderBBalanceAfter.sub(deltaDepositWithdrawal)
+      );
+    })
+
+    .run();
+}
+
+export function ctxFactory(
+  useNewDataFormatOnL1: boolean,
+  useNewDataFormatOnL2: boolean,
+  depositAmount: BigNumber,
+  withdrawalAmount: BigNumber
+) {
+  return async () => {
+    const networkName = env.network("TESTING_OPT_NETWORK", "mainnet");
```

Comment:
Looks weird to have testing and mainnet together tbh

Reply:
It's related to the network name only. Integration tests are running only on forks.